### PR TITLE
chore: js-peer bump deps including libp2p to 1.x

### PR DESCRIPTION
**Pull request by dozyio**

### CODE REVIEW

**Comment by MarcoPolo** [comment]
I can't see the failure log from Vercel. Who has permissions to? @p-shahi do you know?

**Comment by p-shahi** [comment]
```
[13:44:40.169] Running build in Washington, D.C., USA (East) – iad1
[13:44:40.276] Cloning github.com/libp2p/universal-connectivity (Branch: main, Commit: fb04a04)
[13:44:40.590] Cloning completed: 314.037ms
[13:44:40.904] Previous build caches not available
[13:44:41.123] Running "vercel build"
[13:44:41.641] Vercel CLI 33.5.4
[13:44:41.997] Installing dependencies...
[13:44:53.294] npm WARN deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.
[13:44:53.294] npm WARN deprecated @babel/plugin-proposal-numeric-separator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-numeric-separator instead.
[13:44:53.577] npm WARN deprecated @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
[13:44:53.578] npm WARN deprecated @babel/plugin-proposal-async-generator-functions@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-async-generator-functions instead.
[13:44:53.847] npm WARN deprecated @babel/plugin-proposal-optional-catch-binding@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-catch-binding instead.
[13:44:54.198] npm WARN deprecated @babel/plugin-proposal-optional-chaining@7.21.0: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chaining instead.
[13:44:55.683] npm WARN deprecated @babel/plugin-proposal-object-rest-spread@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-object-rest-spread instead.
[13:45:13.025] 
[13:45:13.025] added 1069 packages in 31s
[13:45:13.026] 
[13:45:13.026] 196 packages are looking for funding
[13:45:13.026]   run `npm fund` for details
[13:45:13.069] Detected Next.js version: 14.1.3
[13:45:13.077] Detected `package-lock.json` generated by npm 7+
[13:45:13.077] Running "npm run build"
[13:45:13.493] 
[13:45:13.493] > build
[13:45:13.493] > next build
[13:45:13.493] 
[13:45:14.074] Attention: Next.js now collects completely anonymous telemetry regarding usage.
[13:45:14.074] This information is used to shape Next.js' roadmap and prioritize features.
[13:45:14.075] You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
[13:45:14.075] https://nextjs.org/telemetry
[13:45:14.075] 
[13:45:14.168]    ▲ Next.js 14.1.3
[13:45:14.168] 
[13:45:14.169]    Linting and checking validity of types ...
[13:45:16.542] 
[13:45:16.542] ./src/components/chat.tsx
[13:45:16.543] 405:13  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
[13:45:16.543] 423:13  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
[13:45:16.543] 443:13  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
[13:45:16.543] 
[13:45:16.543] info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
[13:45:19.097] Failed to compile.
[13:45:19.097] 
[13:45:19.097] ./src/pages/index.tsx:5:33
[13:45:19.097] Type error: Cannot find module '@libp2p/interface/connection' or its corresponding type declarations.
[13:45:19.097] 
[13:45:19.097] [0m [90m 3 |[39m [36mimport[39m [33mNav[39m [36mfrom[39m [32m'@/components/nav'[39m[0m
[13:45:19.097] [0m [90m 4 |[39m [36mimport[39m { useLibp2pContext } [36mfrom[39m [32m'@/context/ctx'[39m[0m
[13:45:19.097] [0m[31m[1m>[22m[39m[90m 5 |[39m [36mimport[39m type { [33mConnection[39m } [36mfrom[39m [32m'@libp2p/interface/connection'[39m[0m
[13:45:19.097] [0m [90m   |[39m                                 [31m[1m^[22m[39m[0m
[13:45:19.097] [0m [90m 6 |[39m [36mimport[39m { usePeerContext } [36mfrom[39m [32m'../context/peer-ctx'[39m[0m
[13:45:19.097] [0m [90m 7 |[39m [36mimport[39m { useCallback[33m,[39m useEffect[33m,[39m useState } [36mfrom[39m [32m'react'[39m[0m
[13:45:19.098] [0m [90m 8 |[39m [36mimport[39m [33mImage[39m [36mfrom[39m [32m'next/image'[39m[0m
[13:45:19.138] Error: Command "npm run build" exited with 1
[13:45:19.868] 
```

Here's the failure log @dozyio , thanks for the ping @MarcoPolo

File: js-peer/src/lib/libp2p.ts [hunk 1]
```diff
@@ -67,24 +68,24 @@ export async function startLibp2p() {
...
     services: {
       pubsub: gossipsub({
-        allowPublishToZeroPeers: true,
+        allowPublishToZeroTopicPeers: true,
         msgIdFn: msgIdFnStrictNoSign,
         ignoreDuplicatePublishError: true,
       }),
       dht: kadDHT({
-        protocolPrefix: "/universal-connectivity",
+        protocol: "/universal-connectivity",
```
Comment (achingbrain) [inline js-peer/src/lib/libp2p.ts:76]: If we want to maintain compatibility with existing universal connectivity peers:

```suggestion
        protocol: "/universal-connectivity/kad/1.0.0",
```

Comment (dozyio) [reply js-peer/src/lib/libp2p.ts]: thanks for the review - done

**Review by p-shahi — APPROVED** [review]
thank you! if @achingbrain can give a thumbs up then I can merge this

**Pull request merged**